pkg/config/config.go

```diff
@@ -161,11 +161,20 @@ func (gc *Config) configureProxies(httpProxy, httpsProxy, noProxy string) {
 	}
 
 	if noProxy != "" {
+		noProxy = addKubernetesServiceHostInNoProxy(noProxy)
 		os.Setenv(noProxyEnvVar, noProxy)
 		os.Setenv(strings.ToUpper(noProxyEnvVar), noProxy)
 	}
 }
 
+func addKubernetesServiceHostInNoProxy(noProxy string) string {
+	if strings.Contains(noProxy, "KUBERNETES_SERVICE_HOST") {
+		k8sSvcHost := os.Getenv("KUBERNETES_SERVICE_HOST")
```

Comment:
do we need to augment this with any other "shorthands", like kubernetes-api.kube-system.svc.local. etc? might we worth asking folks that are raising this issue.

Reply:
cc @ssamrit-vmw 

Anything else we should consider here via this shorthand approach? Any preferred UX for it?

Reply:
Shouldn't we first split the noProxy with comma delimiter and then check for KUBERNETES_SERVICE_HOST.
With current implementation if someone provides "$(KUBERNETES_SERVICE_HOST)" in noProxy then it will end up creating "$(<ip>)", right?

>Anything else we should consider here via this shorthand approach? Any preferred UX for it?

In our testing of kapp-controller on cluster configured with proxy using TMC, we found that adding this ip was sufficient for kapp-controller to work.
We have package-deployment-extension deployed on cluster which will set this value in noProxy of kapp config if cluster is configured with proxy configuration using TMC, so we don't expect user to set it while creating proxy configuration in TMC.

Reply:
If someone provides "$(KUBERNETES_SERVICE_HOST)" it will be "$(10.x.y.z)" which I think is a much clearer error that someone made a mistake than if they had put "$(KUBERNETES_SERVICE_HOST)" and got "$(KUBERNETES_SERVICE_HOST)" because it doesn't match 1:1.

Reply:
@neil-hickey I think the challenge here is when is when is this information surfaced to the user. The user won't understand how kapp-controller is replacing `KUBERNETES_SERVICE_HOST` and it will never be displayed in the associated controller secret. We should maybe consider having some kind of more direct error returned saying the shorthand provided was not an exact match.

Reply:
makes sense. I figured this information would be surfaced in some way, beyond just the logs. If that's not really the case, and they supply "xyz-KUBERNETES_SERVICE_HOST" or anything in addition to just the string "KUBERNETES_SERVICE_HOST" then an error seems reasonable, given we are manipulating the input in this particular case.

Reply:
So this is not perfect in the sense that this will only protect against extra characters, but it will provide some failure information to help a user navigate this scenario. Hopefully noticing the noProxy not being properly configured / being able to notice the typo will provide enough detail in the case of a typo, etc.